lib/IRGen/GenDecl.cpp

```diff
@@ -644,6 +665,31 @@ void IRGenModule::addProtocolConformanceRecord(
   ProtocolConformances.push_back(conformance);
 }
 
+static bool
+protocolConformanceExistsForType(
+  SmallVectorImpl<NormalProtocolConformance *> &ProtocolConformances, CanType type) {
+  for (auto *conformance : ProtocolConformances) {
+    auto conformanceType = conformance->getType()->getCanonicalType();
```

Comment:
This is pretty inefficient. You should be able to ask a `NominalTypeDecl` what protocols it conforms to. You'll also have to filter out `AnyObject` and `@objc` protocols, which we don't emit Swift protocol conformance records for.